libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs

```diff
@@ -111,14 +129,18 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             dc.State.SetValue(TURN_COUNT_PROPERTY, 0);
             dc.State.SetValue(INPUT_PROPERTY, null);
 
-            var state = this.AlwaysPrompt ? InputState.Missing : await this.RecognizeInput(dc, false);
+            var state = this.AlwaysPrompt ? InputState.Missing : await this.RecognizeInput(dc);
             if (state == InputState.Valid)
             {
                 var input = dc.State.GetValue<object>(INPUT_PROPERTY);
                 return await dc.EndDialogAsync(input);
             }
             else
             {
+                // turnCount should increase here, because you want when nextTurn comes in
+                // the value should already be 1, otherwise, this input will pick from "dialog.value"
+                // instead of "turn.activity.text"
+                dc.State.SetValue(TURN_COUNT_PROPERTY, 1);
```

Comment:
Minor detail: The comment makes it sound like you are going to increase the turn count but hwat you do in the code is set it to 1.